backend/inspirehep/pidstore/providers/bai.py

```diff
@@ -88,39 +94,25 @@ def create(
         pid_value = pid_value or get_first_value_for_schema(
             get_value(data, "ids", []), "INSPIRE BAI"
         )
-        retry_count = (
-            current_app.config.get("PIDSTORE_BAI_MAX_RETRY_COUNT", 5)
-            if not pid_value
-            else 1
-        )
-        for _ in range(retry_count):
-            last_exception = None
-            new_pid = pid_value or cls.generate_bai(data)
-            pid_from_db = cls.query_pid_value(new_pid)
-            if not pid_from_db:
-                try:
-                    provider_object = super().create(
-                        pid_value=new_pid,
-                        object_type=object_type,
-                        object_uuid=object_uuid,
-                        status=PIDStatus.REGISTERED,
-                        **kwargs,
-                    )
-                    # Pid created successfully.
-                    break
-                except IntegrityError as e:
-                    last_exception = e
-            elif pid_from_db.object_uuid != object_uuid:
-                last_exception = PIDAlreadyExistsError(
-                    pid_value=pid_value, pid_type="bai"
+        new_pid = pid_value or cls.generate_bai(data)
+        pid_from_db = cls.query_pid_value(new_pid)
+        if not pid_from_db:
+            try:
+                provider_object = super().create(
+                    pid_value=new_pid,
+                    object_type=object_type,
+                    object_uuid=object_uuid,
+                    status=PIDStatus.REGISTERED,
+                    **kwargs,
                 )
-            else:
-                provider_object = super().get(pid_value=pid_value)
-                break
-                # Correct pid already assigned to this record
-            time.sleep(current_app.config.get("PIDSTORE_BAI_RETRY_DELAY", 5))
```

Comment:
Is this variable defined in the config? if so you should remove it there too.

Reply:
No, it's not, I'll drop it 👍